### PR TITLE
Add app layer to exceptions of no-layer-public-api rule

### LINKED ISSUE #55: Add an exception for an index file inside the App layer
Right now it reports an error on such a case. But it looks like it should not, because the app layer is the entry point to your application, and it's pretty intuitive that it could have an index.js/ts file.

Screenshot 2024-07-13 at 23 35 16

### PR DESCRIPTION
**Pull request by daniilsapa**
Resolves #55

### CODE REVIEW

**Comment by daniilsapa** [comment]
@illright 
Could you tell me how I can test Steiger locally? How do you usually do that? I tried to run `turbo build` and then run dist/cli.js file but got an error.

**Comment by illright** [comment]
Weird, what error did you get?

File: packages/steiger-plugin-fsd/src/no-layer-public-api/README.md [hunk 1]
```diff
@@ -11,6 +11,8 @@ According to the _public API rule on slices_:
 
 A corollary to this rule is that the layer itself should not have an index file.
 
+**Exception:** index files are allowed on `app` layer because it's the entry point of the application.
```
Comment (illright) [inline packages/steiger-plugin-fsd/src/no-layer-public-api/README.md:14]: I would rephrase this to say "because some people prefer to have that as their app's entrypoint" to emphasize that this is not a rule, but a preference

File: .changeset/fair-ducks-hope.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@feature-sliced/steiger-plugin': minor
```
Comment (illright) [inline .changeset/fair-ducks-hope.md:2]: I think this qualifies more as a patch change, since it's not adding new features

File: packages/steiger-plugin-fsd/src/no-layer-public-api/index.ts [hunk 3]
```diff
@@ -1,6 +1,9 @@
 import { getIndex, getLayers } from '@feature-sliced/filesystem'
 import type { Diagnostic, Rule } from '@steiger/types'
 
+/** Layers that are allowed to have an index file. */
```
Comment (illright) [inline packages/steiger-plugin-fsd/src/no-layer-public-api/index.ts:4]: praise: thanks for appropriately documenting your code!

**Review by illright — CHANGES_REQUESTED** [review]
Another great PR, thanks! A few corrections from my side

File: .changeset/fair-ducks-hope.md (hunk 2 above)
Comment (daniilsapa) [reply .changeset/fair-ducks-hope.md]: Indeed, you’re right

**Comment by daniilsapa** [comment]
The error occurs when you try to specify a folder that is outside the working folder. This is true for both local and distributed app builds.

Here's what it looks like

Screenshot 2024-07-17 at 16 23 09

**Review by illright — APPROVED** [review]
Thanks for implementing the corrections! As for the issues you're facing in development, it's weird, I've never seen it before. Could you make an issue about it, so that we don't lose this conversation?

**Comment by daniilsapa** [comment]
You're welcome. Yes, I created an issue for that error
